libs/web/src/main/kotlin/net/corda/web/server/JavalinServer.kt

```
@@ -0,0 +1,111 @@
+package net.corda.web.server
+
+import io.javalin.Javalin
+import net.corda.lifecycle.LifecycleCoordinatorFactory
+import net.corda.lifecycle.LifecycleStatus
+import net.corda.lifecycle.createCoordinator
+import net.corda.utilities.classload.executeWithThreadContextClassLoader
+import net.corda.utilities.executeWithStdErrSuppressed
+import net.corda.v5.base.exceptions.CordaRuntimeException
+import net.corda.web.api.Endpoint
+import net.corda.web.api.HTTPMethod
+import net.corda.web.api.WebServer
+import org.eclipse.jetty.websocket.servlet.WebSocketServletFactory
+import org.osgi.framework.FrameworkUtil
+import org.osgi.framework.wiring.BundleWiring
+import org.osgi.service.component.annotations.Activate
+import org.osgi.service.component.annotations.Component
+import org.osgi.service.component.annotations.Reference
+import org.slf4j.LoggerFactory
+
+
+@Component(service = [WebServer::class])
+class JavalinServer @Activate constructor(
+    @Reference(service = LifecycleCoordinatorFactory::class)
+    private val coordinatorFactory: LifecycleCoordinatorFactory,
+    @Reference(service = JavalinFactory::class)
+    private val javalinFactory: JavalinFactory
+) : WebServer {
+
+    private companion object {
+        val log = LoggerFactory.getLogger(this::class.java.enclosingClass)
+    }
+
+    private var server: Javalin? = null
+    private val coordinator = coordinatorFactory.createCoordinator<WebServer> { _, _ -> }
+    private val endpoints: MutableList<Endpoint> = mutableListOf()
+
+    override fun start(port: Int) {
+        if (server != null) {
+            throw CordaRuntimeException("The Javalin webserver is already initialized")
+        }
```

Do we need to throw an exception here? Could we just return instead?

---

Addressed in https://github.com/corda/corda-runtime-os/pull/4631